src/components/Chat/ChatAction.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as cx from 'classnames'
+
+import {
+  childrenExist,
+  createShorthandFactory,
+  customPropTypes,
+  IRenderResultConfig,
+  UIComponent,
+} from '../../lib'
+import {
+  ComponentPartStyle,
+  ComponentVariablesInput,
+  IComponentPartStylesInput,
+} from '../../../types/theme'
+import { Extendable, ItemShorthand, ReactChildren } from '../../../types/utils'
+import Layout from '../Layout'
+import Text from '../Text'
+import Icon from '../Icon'
+
+export interface IChatActionProps {
+  as?: any
+  icon?: ItemShorthand
+  children?: ReactChildren
+  className?: string
+  content?: any
+  styles?: ComponentPartStyle
+  timestamp?: ItemShorthand
+  variables?: ComponentVariablesInput
+}
+
+class ChatAction extends UIComponent<Extendable<IChatActionProps>, any> {
+  static className = 'ui-chat__action'
+
+  static create: Function
+
+  static displayName = 'ChatAction'
+
+  static propTypes = {
+    as: customPropTypes.as,
+
+    /** Control chat messages can have an icon */
+    icon: customPropTypes.itemShorthand,
+
+    /** Child content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Shorthand for the primary content. */
+    content: PropTypes.any,
+
+    /** Custom styles to be applied for component. */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Timestamp of the message. */
+    timestamp: customPropTypes.itemShorthand,
+
+    /** Custom variables to be applied for component. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  }
+
+  static handledProps = [
+    'as',
+    'children',
+    'className',
+    'content',
+    'icon',
+    'styles',
+    'timestamp',
+    'variables',
+  ]
+
+  static defaultProps = {
+    as: 'div',
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    rest,
+    styles,
+    variables,
+  }: IRenderResultConfig<IChatActionProps>) {
+    const { as, icon, children } = this.props
+
+    return childrenExist(children) ? (
```

Comment:
@mnajdova - it looks like the `Chat.Action` subcomponent is more complex (per @codepretty 's comments in our chats, we can have an array of actions with more complicated logic on what's rendered); I'm not even sure it falls under Stardust principles, maybe we should be more generic with the `action` prop in `Chat` component

Reply:
In my opinion, we shouldn't alter this component at this moment. We can reason for this in the following manner:
- Chat.Action can have children which can define whatever the user wants inside - in this case we won't change anything right now
- We can add actions as a list and make the Chat.Action use the Accordion component, but in that case it would be in my opinion very much Team's specific rather then common web (as @Bugaa92 said, it doesn't really fall under stardust principles)

My final thought on this is, leaving the Chat.Action component as is for this now, and address this later if we think would be necessary. First, I would like after this is merged, to try and address this in the chat pane and we will see how much complication it would bring.